### PR TITLE
update kube policy to drop tunnel

### PR DESCRIPTION
**Pull request by sidpremkumar**
## Description of the change

Removes refs to kube tunnel and update to kube

See related common pr: https://github.com/bastionzero/webshell-common-ts/pull/52

## Relevant release note information

Release Notes:

## Related JIRA tickets

Relates to JIRA: CWC-XXX

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [x] No

If yes, please explain:

### CODE REVIEW

File: src/handlers/group/add-group-policy.handler.v2.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export async function addGroupToPolicyHandler(groupName: string, policyName: str
     const policy = kubePolicy ? kubePolicy : targetPolicy;
 
     // If this group exists already
-    const group = policy.groups.find(g => g.name == groupSummary.name);
+    const group = policy.groups.find((g: any) => g.name == groupSummary.name);
```
Comment (chad-cwc) [inline src/handlers/group/add-group-policy.handler.v2.ts:43]: Was this giving an error or warning before you changed it to `any`? Can it be of type `Group` from http/v2/policy/types/group.types.ts?

Comment (sidpremkumar) [reply src/handlers/group/add-group-policy.handler.v2.ts:43]: it was giving a warning before. I will update to group

File: src/handlers/policy/list-kubernetes-policies.handler.ts [hunk 2]
```diff
@@ -60,14 +60,14 @@ export async function listKubeTunnelPoliciesHandler(
 
     if(!! argv.json) {
         // json output
-        console.log(JSON.stringify(kubeTunnelPolicies));
+        console.log(JSON.stringify(kubernetesPolicies));
     } else {
-        if (kubeTunnelPolicies.length === 0){
+        if (kubernetesPolicies.length === 0){
             logger.info('There are no available Kubernetes Tunnel policies');
```
Comment (chad-cwc) [inline src/handlers/policy/list-kubernetes-policies.handler.ts:66]: Remove "Tunnel". Probably need to do a project-wide case-insensitive Find on "tunnel".

File: src/services/v1/policy/policy.types.ts [hunk 3]
```diff
@@ -18,7 +18,9 @@ export enum PolicyType {
     TargetConnect = 'TargetConnect',
     OrganizationControls = 'OrganizationControls',
     SessionRecording = 'SessionRecording',
-    KubernetesTunnel = 'KubernetesTunnel'
+    KubernetesTunnel = 'KubernetesTunnel', // Deprecated in favor of Kubernetes
```
Comment (chad-cwc) [inline src/services/v1/policy/policy.types.ts:21]: I'm trying to think of any reason not to just remove this. Truly deprecating something usually means it's still around in the code but shouldn't be used for new code. I think that's probably not the case here, right?

Comment (sidpremkumar) [reply src/services/v1/policy/policy.types.ts:21]: Its an enum, and we cannot delete the enum in our backend (SQL limitation I believe), so we keep it around here so the Kubernetes enum maps to the correct int value

Comment (sidpremkumar) [reply src/services/v1/policy/policy.types.ts:21]: Its an enum in our backend, https://github.com/bastionzero/webshell-backend/blob/feat/db-support/Webshell.Database/Database/Schema/Types/PolicyType.cs#L7-L17

And we cannot remove it as there is an SQL limitation that blocks us from this (not 100% but I know we have had issues deleting enums before and have used Obsolete). 

So in order to make sure the mapping works, we keep this around

Comment (sidpremkumar) [reply src/services/v1/policy/policy.types.ts:21]: https://github.com/bastionzero/webshell-backend/blob/2fe6627f685b5f93fa4ee9955c7ffbce60840a3d/Webshell.Database/Database/Schema/Types/ConnectionType.cs#L11-L13

Comment (chad-cwc) [reply src/services/v1/policy/policy.types.ts:21]: But will the back end ever return the obsolete enum value? (I'm assuming a DB migration will change all old values to the new one, but maybe I'm wrong.) If the back end will never return the obsolete value, there is no reason I can think of to keep the client code in sync with the server code.

Of course, it existing policies will use the obsolete PolicyType value, then we must keep it here.

Comment (sidpremkumar) [reply src/services/v1/policy/policy.types.ts:21]: From my understanding these PolicyTypes map to this on the backend: https://github.com/bastionzero/webshell-backend/blob/feat/db-support/Webshell.Database/Database/Schema/Types/PolicyType.cs#L2-L17

So we will never return `KubernetesTunnel` but then how can I make sure that `Kubernetes` maps to the correct enum (i.e. `4`)?

Im not sure of a way to delete / omit a value from an enum type in js and preserver int relationship. For example if I updated this to: 
```js
export enum PolicyType {
     TargetConnect = 'TargetConnect',
     OrganizationControls = 'OrganizationControls',
     SessionRecording = 'SessionRecording',
     Kubernetes = 'Kubernetes',
     Proxy = 'Proxy'
 }
```

Now Kuberneters will map to the enum `3` which is incorrect on our backend.

**Comment by EthanHeilman** [comment]
@chad-cwc Does this PR have any security impact?

**Comment by sidpremkumar** [comment]
> @chad-cwc Does this PR have any security impact?

No but its not a PR into master, it's just a PR into a feature branch.

File: src/handlers/target-group/list-target-group.handler.v2.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ import { PolicyHttpService } from '../../../src/http-services/policy/policy.http
 export async function listTargetGroupHandler(configService: ConfigService, logger: Logger, argv : yargs.Arguments<targetGroupArgs>, policyName: string) {
 
     const policyHttpService = new PolicyHttpService(configService, logger);
-    const kubePolicies = await policyHttpService.ListKubeTunnelPolicies();
+    const kubePolicies = await policyHttpService.ListKubernetesPolicies();
```
Comment (lgmugnier) [inline src/handlers/target-group/list-target-group.handler.v2.ts:12]: There's something about the fact that KubeTunnel takes up the same number of pixels as Kubernetes that is *chef's kiss*

**Review by lgmugnier — APPROVED** [review]

File: src/services/v1/policy/policy.types.ts (hunk 3 above)
Comment (sidpremkumar) [reply src/services/v1/policy/policy.types.ts:21]: We are in the middle of trying to get everything synced up so I went ahead and merged this into the feature branch. Please feel free to comment on this PR and I will make the changes in a follow up PR.

**Pull request merged**